src/BaseStrategy.js

```
@@ -0,0 +1,80 @@
+     * @param {string} address - The wallet address.
+     * @param {string} serviceProvider - The name of the service provider.
+     * @param {string} userId - The user's unique identifier.
+     * @param {string} username - The user's username.
+     * @returns {string} The verification URL.
+     * @throws {ErrorWithMessage} Throws an error if the address or service provider is invalid.
+     */
+    getVerifyUrl(address, serviceProvider, userId, username) {
+        if (Validation.isWalletAddress(address) && Validation.isServiceProvider(serviceProvider) && Validation.isUserId(userId, serviceProvider) && Validation.isUserName(username)) {
+            return `${process.env.domain}/verify/${serviceProvider}/${address}/${String(userId)}/${encodeURIComponent(String(username).toLowerCase())}`;
+        } else {
+            throw new ErrorWithMessage('Invalid Data', { address, serviceProvider, address, userId, code: 'INVALID_DATA' });
```

Biome (1.9.4)

[error] This shorthand property named address is later overwritten by an object member with the same name.

Overwritten with this value.

If an object property with the same name is defined multiple times (except when combining a getter with a setter), only the last definition makes it into the object and previous definitions are ignored.
Unsafe fix: Remove this shorthand property named address


(lint/suspicious/noDuplicateObjectKeys)